tests/svg.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const {DOMParser} = require('xmldom');
+const xpath = require('xpath');
+const core = require('@actions/core');
+let errors = false;
+
+// Function to test the SVG content against an XPath expression
+function test(svgContent, xpathExpression) {
+  try {
+    const doc = new DOMParser().parseFromString(svgContent, 'application/xml');
+    const nodes = xpath.select(xpathExpression, doc);
+    return nodes.length > 0;
+  } catch (err) {
+    core.error(`Failed to parse SVG content: ${err.message}`);
+    return false;
+  }
+}
+
+// Function to handle file checking
+async function main() {
+  const files = process.argv.slice(2);
+
+  await Promise.allSettled(files.map(async (file) => {
+    const error = (msg) => {
+      core.error(msg, {file});
+      errors = true;
+    };
+
+    const warn = (msg) => {
+      core.warning(msg, {file});
+    };
+
+    const svg = fs.readFileSync(file, 'utf8');
+    const doc = new DOMParser().parseFromString(svg, 'application/xml');
+    const parseErrors = doc.getElementsByTagName('parsererror');
+
+    if (parseErrors.length > 0)
+      error('Invalid SVG file');
+    if (svg.includes('<?'))
+      error('Unnecessary processing instruction found');
+    if (test(svg, '//image'))
+      error('Embedded image detected');
+    if (svg.split('\n').filter(line => line.trim()).length > 1)
+      error('Minimize file to one line');
+    if (test(svg, '//comment()'))
+      warn('Remove comments');
+    if (fs.statSync(file).size > 5 * 1024)
+      warn('Unusually large file size');
```

Comment:
Would it be beneficial to say something like "Unusually large file size, try running this file through SVGOMG" or would that be a useless addition?

Reply:
SVGOMG certainly helps, and sometimes, that's all that's needed, but usually, this test failing indicates that the icon isn't meant to be an SVG, like some American universities using coats of arms or Swedbank's [icon of an oak tree](https://www.swedbank.com/newsroom/logotype.html).

Since it isn't clear if the SVG file just needs to be run through SVGOMG (like with my screenshot below) or if it is unsuitable as an SVG, I think it is better to not mention SVGOMG in the warning message.

![SVGOMG](https://github.com/user-attachments/assets/463122e1-c028-4fb6-a636-e9181c9a7300)

Reply:
Sounds good to me.